src/test/java/com/shapesecurity/salvation/ParserTest.java

```diff
@@ -1050,4 +1050,34 @@
         assertEquals(1, p.getDirectives().size());
         assertEquals(0, notices.size());
     }
+
+    @Test public void testReportSample() {
+        Policy p;
+        ArrayList<Notice> notices = new ArrayList<>();
+        p = parseWithNotices("default-src 'report-sample'", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals(0, notices.size());
+
+        notices.clear();
+        p = parseWithNotices("script-src 'report-sample' 'report-sample'", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals("script-src 'report-sample'", p.getDirectiveByType(ScriptSrcDirective.class).show());
+        assertEquals(0, notices.size());
+
+        notices.clear();
+        p = parseWithNotices("style-src 'report-sample' 'unsafe-hashed-attributes' 'report-sample' 'nonce-123'", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals(2, notices.size());
+        assertEquals("Invalid base64-value (should be multiple of 4 bytes: 3).", notices.get(0).message);
```

Comment:
You have two notices and the one you're asserting on is unrelated to report-sample. Just fix the nonce and assert on the remaining notice.